tensorflow/compiler/xla/service/algebraic_simplifier.cc

```diff
@@ -2520,6 +2523,13 @@ Status AlgebraicSimplifierVisitor::HandleTranspose(HloInstruction* transpose) {
                                            transpose->dimensions())));
   }
 
+  if (ShapeUtil::TrueRank(transpose->shape()) <= 1) {
```

Comment:
Can you generalize this to Detect degenerate dimension, strip them from the transpose and then make remove ones reshape, transpose, add ones reshape. At that point the tranpose becomes a no-op and the reshapes are free to merge in the next fixed point through the algebraic simplifier.